### ISSUE TITLE
Cop Suggestion: warn against usage of `update_attribute` in Rails

### ISSUE THREAD

**Issue by rahulcs** [bbatsov/rubocop#3743]
The [update_attribute](http://api.rubyonrails.org/classes/ActiveRecord/Persistence.html#method-i-update_attribute) method skips validations and will save the object to database regardless of its validity.
[These](http://guides.rubyonrails.org/active_record_validations.html#skipping-validations) methods should be used with caution.

Prefer this:
```ruby
user.update_attributes(email: 'example@example.com')
```
over:
```ruby
user.update_attribute(email: 'example@example.com')
```

Bugs resulting from this can be hard to identify, and I cannot see a valid use-case for updating a model attribute without validation.

Is this a reasonable candidate for inclusion as a Rails-specific cop in RuboCop?

**Comment by tejasbubane** [comment]
Not just `update_attribute`, I think we should we warn against [all of those methods](http://guides.rubyonrails.org/active_record_validations.html#skipping-validations) which skip validations. Even `save(validate: false)`. 

> Is this a reasonable candidate for inclusion as a Rails-specific cop in RuboCop?

Sounds like a pretty reasonable candidate 👍  Let's see what others have to say.

**Comment by rrosenblum** [comment]
> Not just `update_attribute`, I think we should we warn against all of those methods which skip validations. Even `save(validate: false)`.

I agree that we should warn against all of the methods that skip validations. I don't think that we should warn against `save(validate: false)` because someone has to explicitly say that they do not want validations. The other ActiveRecord methods are most likely used by mistake.
